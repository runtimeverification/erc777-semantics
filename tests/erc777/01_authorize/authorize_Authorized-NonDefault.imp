test.preamble();

// Testing authorizeOperator(Operator) when Operator is an already authorized operator.
// A token holder MAY authorize an already authorized operator. An AuthorizedOperator MUST be emitted each time.

test.isOperatorFor(1, 7);
test.authorizeOperator(1);
test.isOperatorFor(1, 7);
test.authorizeOperator(1);
test.isOperatorFor(1, 7);
