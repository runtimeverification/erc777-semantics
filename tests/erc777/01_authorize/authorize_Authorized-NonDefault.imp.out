1 is not operator for 7
7 is authorizing 1 as an operator
1 is operator for 7
7 is authorizing 1 as an operator
1 is operator for 7
